packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -141,12 +141,12 @@ export class SummaryManager implements IDisposable {
         // enforce connectedState.clientId === clientElection.electedClientId. But once we're Running, we should
         // only transition to Stopping when the electedParentId changes. Stopping the summarizer without
         // changing the electedParent will just cause us to transition to Starting again.
-        if (!this.connectedState.connected) {
-            return { shouldSummarize: false, stopReason: "parentNotConnected" };
-        } else if (this.connectedState.clientId !== this.clientElection.electedParentId ||
+        if (this.connectedState.clientId !== this.clientElection.electedParentId ||
             (this.state !== SummaryManagerState.Running &&
                 this.connectedState.clientId !== this.clientElection.electedClientId)) {
             return { shouldSummarize: false, stopReason: "parentShouldNotSummarize" };
+        } else if (!this.connectedState.connected) {
+            return { shouldSummarize: false, stopReason: "parentNotConnected" };
         } else if (this.disposed) {
```

Comment:
@agarwal-navin can you please double check my logic in #9946? You already looked at it, but I want to make sure the reasoning is solid on why this change is correct and preferred.

Reply:
Taking a look. What's the reason for re-ordering the first 2 conditions though?

Reply:
Looks reasonable. My only questions is the one I asked above. This one is not clear to me.